packages/source-iottwinmaker/src/knowledgeGraph-module/KGDataModule.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  ExecuteQueryCommand,
+  ExecuteQueryCommandOutput,
+  ExecuteQueryCommandInput,
+  IoTTwinMakerClient,
+  ColumnDescription,
+  Row,
+} from '@aws-sdk/client-iottwinmaker';
+import { KGDataModuleInput } from './types';
+import { TwinMakerKGQueryDataModule } from '../types';
+
+export class KGDataModule implements TwinMakerKGQueryDataModule {
+  private twinMakerClient: IoTTwinMakerClient;
+  private workspaceId: string;
+
+  constructor(input: KGDataModuleInput) {
+    this.twinMakerClient = input.twinMakerClient;
+    this.workspaceId = input.workspaceId;
+  }
+
+  executeQuery = async (queryStatement: string, maxResultsCount?: number): Promise<ExecuteQueryCommandOutput> => {
+    const rows: Row[] = [];
+    let columnDescriptions: ColumnDescription[] = [];
+    let $metadata = {};
+    const commandInput: ExecuteQueryCommandInput = {
+      workspaceId: this.workspaceId,
+      queryStatement: queryStatement,
+    };
+    do {
+      const response: ExecuteQueryCommandOutput = await this.twinMakerClient.send(
+        new ExecuteQueryCommand(commandInput)
+      );
+      if (response.rows) {
+        rows.push(...response.rows);
+      }
+      if (response.columnDescriptions && columnDescriptions.length === 0) {
+        columnDescriptions = response.columnDescriptions;
+      }
+      if (response.$metadata && Object.keys($metadata).length === 0) $metadata = response.$metadata;
+      commandInput.nextToken = response.nextToken;
+    } while (maxResultsCount ? commandInput.nextToken && maxResultsCount : commandInput.nextToken);
```

Comment:
I think this line reduces to just:

} while (commandInput.nextToken);

which means maxResultsCount can be dropped as a parameter as it has no impact?

Reply:
+1

Reply:
oops what I meant is rows.length < maxResultsCount
 (maxResultsCount ? commandInput.nextToken && rows.length < maxResultsCount : commandInput.nextToken)
I fixed it.